das/simple_das_reader_aggregator.go

```diff
@@ -275,5 +275,11 @@ func (a *SimpleDASReaderAggregator) String() string {
 }
 
 func (a *SimpleDASReaderAggregator) HealthCheck(ctx context.Context) error {
+	for _, serv := range a.readers {
+		err := serv.HealthCheck(ctx)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
I don't think a health check for the aggregators that has the logic "if any backend fails, fail the aggregator health check" makes sense because backends will fail from time-to-time but the aggregator itself will still function.

It doesn't really matter for the `das.Aggregator` above since we won't be recommending people run it on their `daserver` deployments, but the `das.SimpleDASReaderAggregator` will be run in everyone's deployment as the backup for the `FallbackStorageService`. The list of URLs for the `das.SimpleDASReaderAggregator`  is likely to be long, and it learns what backends are the responsive ones over time so it can tolerate some backends being down quite well.

So I think at least for the `das.SimpleDASReaderAggregator`  this change should be reverted.

Reply:
Makes sense, reverted to nil.